text/0000-let-else.md

```diff
@@ -242,16 +242,39 @@ a lazy boolean operator is present (`&&` or `||`), for reasons noted in [future-
 
 If the pattern is irrefutable, rustc will emit the `irrefutable_let_patterns` warning lint, as it does with an irrefutable pattern in an `if let`.
 
-The `else` block must diverge. This could be a keyword which diverges (returns `!`), or a panic.
-This likely necessitates a new subtype of `BlockExpression`, something like `BlockExpressionDiverging`.
-Allowed keywords:
-- `return`
-- `break`
-- `continue`
+The `else` block must _diverge_, meaning the `else` block must return the [never type (`!`)][never type]).
+This could be a keyword which diverges (returns `!`), or a panic.
 
 If the pattern does not match, the expression is not consumed, and so any existing variables from the surrounding scope are
 accessible as they would normally be.
 
+For patterns which match multiple variants, such as through the `|` (or) syntax, all variants must produce the same bindings (ignoring additional bindings in uneven patterns),
+and those bindings must all be names the same. Valid example:
+```rust
+let Some(x) | MyEnum::VariantA(_, _, x) | MyEnum::VariantB { x, .. } = a else { return; };
```

Comment:
These patterns have different types. Is that intentional?

Reply:
Oh those all need to be `MyEnum`, whoops.